app/services/variant_deleter.rb

```diff
@@ -11,6 +11,17 @@ def delete(variant)
     variant.destroy
   end
 
+  def destroy_related_outgoing_variants(variant_id, receiver)
+    parent_variants = ExchangeVariant.where(variant_id: variant_id)
+    internal_variants = ExchangeVariant.
+      joins(:exchange).
+      where(
+        exchange_variants: { variant_id: parent_variants.pluck(:variant_id) },
+        exchanges: { sender: receiver, receiver: receiver, incoming: false }
+      )
```

Comment:
I shouldn't look at this at the end of my work day. What are the parent variants variants here? Aren't you selecting ExchangeVariants by variant_id and then plucking that variant_id into selecting ExchangeVariants again? Would it be the same as this?
```suggestion
    internal_variants = ExchangeVariant.where(variant_id: variant_id)
      joins(:exchange).
      where(
        exchanges: { sender: receiver, receiver: receiver, incoming: false }
      )
```